libflux/flux-core/src/semantic/types.rs

```diff
@@ -109,21 +110,53 @@ impl PartialEq for PolyType {
             kinds.sort();
         }
 
+        assert_eq!(a.vars, b.vars);
+        assert_eq!(a.cons, b.cons);
+        assert_eq!(a.expr, b.expr);
```

Comment:
Maybe you added these assertions for debugging? it seems like you really just want this function to return the expression just below?

Reply:
Woops, fixed.